File: Practice_Algorithm/876. Middle of the Linked List.java
class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode s = head, f = head;
        while(f.next != null && f.next.next != null){
            s = s.next;
            f = f.next.next;
        }
        return f.next == null?s:s.next;
    }
}

File: PastYears/Leetcode 75/876. Middle of the Linked List.cpp
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        ListNode *slow = head, *fast = head;
        while(slow->next && fast->next && fast->next->next){
            slow = slow->next;
            fast = fast->next->next;
        }
        return fast->next == NULL ? slow : slow->next;
    }
        
};
